src/main/java/org/opensearch/knn/index/query/lucenelib/NestedKnnVectorQueryFactory.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.query.lucenelib;
+
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.join.BitSetProducer;
+import org.apache.lucene.search.join.DiversifyingChildrenByteKnnVectorQuery;
+import org.apache.lucene.search.join.DiversifyingChildrenFloatKnnVectorQuery;
+import org.opensearch.knn.index.query.common.QueryUtils;
+
+/**
+ * A class to create a nested knn vector query for lucene
+ */
+public class NestedKnnVectorQueryFactory {
+    /**
+     * Create a query for k-NN nested field.
+     *
+     * The query is generated two times when inner_hits() parameter exist in the request.
+     * For inner hit, we return all filtered nested field documents belongs to the final result of parent documents.
```

Comment:
Ive been thinking this over and it does not seem correct. By passing inner_hits, we are altering the behavior of the query. One of the reasons for inner_hits is to understand why certain results from a query are returned (https://opensearch.org/docs/latest/search-plugins/searching-data/inner-hits/#benefits-of-using-inner_hits). By making this the behavior, we are altering the logic and thus compromising the integrity of the feature from a debugging perspective - you could get different results depending on if you pass "inner_hits" or not.

That being said, I think we can expand the other children docs but just mark the scores as null to signal that the results exist, but are not matches. See result of https://opensearch.org/docs/latest/search-plugins/searching-data/inner-hits/#example-inner_hits-parameters-with-a-parent-join-relationship in 3. In this approach, users can do what they want with all of the hits, but does not alter the behavior.

Reply:
> The main query results will remain unchanged regardless of whether `inner_hits` is included.

Will scores differ?

> Having scores is what users expect. Without scores, `inner_hits` adds limited value.

Yes, but users also expect that the `inner_hits` show how the parent score got formed. So, it is misleading if the result 1 has score = child_A_Score, but child_B_Score > child_A_score. The question would then be why did the parent have a score lower than its greatest child.

Reply:
If users want scores of all children, we should introduce some parameter like expand children or add something in the rescore section on children. I dont think we should do it by default unless we change the baseline functionality of the main query behavior

Reply:
> Will scores differ?
Score of main query will remain same.

> Yes, but users also expect that the inner_hits show how the parent score got formed. So, it is misleading if the result 1 has score = child_A_Score, but child_B_Score > child_A_score. The question would then be why did the parent have a score lower than its greatest child.

That is valid point and we will document it why it happens. 

> If users want scores of all children, we should introduce some parameter like expand children or add something in the rescore section on children. I dont think we should do it by default unless we change the baseline functionality of the main query behavior

That's a valid point, and I like the idea. Perhaps we could introduce a new parameter in the KNN query to specify whether to expand children or not, ensuring consistent behavior between `inner_hits` and non-`inner_hits` queries? This way, they can also utilize the `score_mode` if they want without breaking backward compatibility.

```
{
  "query": {
    "nested": {
      "path": "nested_field",
      "query": {
        "knn": {
          "nested_field.my_vector": {
            "vector": [
              6,
              6,
              6
            ],
            "k": 2,
            "rescore_with_nested_docs": true
          }
        }
      }
    }
  }
}
```